internal/runners/packages/searchTable.go

```diff
@@ -0,0 +1,222 @@
+	}
+
+	doc := strings.Builder{}
+	for _, pkg := range s.Results {
+		if pkg.Name != "" {
+			doc.WriteString(formatRow(styleBold.Render(keyName), pkg.Name, maxKeyLength, s.width))
+		}
+		if pkg.Description != "" {
+			doc.WriteString(formatRow(styleBold.Render(keyDescription), pkg.Description, maxKeyLength, s.width))
+		}
+		if pkg.Website != "" {
+			doc.WriteString(formatRow(styleBold.Render(keyWebsite), styleCyan.Render(pkg.Website), maxKeyLength, s.width))
+		}
+		if pkg.License != "" {
+			doc.WriteString(formatRow(styleBold.Render(keyLicense), pkg.License, maxKeyLength, s.width))
+		}
+
+		var versions []string
+		for i, v := range pkg.Versions {
+			if i > 5 {
+				versions = append(versions, locale.Tl("search_more_versions", "... ({{.V0}} more)", strconv.Itoa(len(pkg.Versions)-5)))
+				break
+			}
+			versions = append(versions, styleCyan.Render(v))
+		}
+		if len(versions) > 0 {
+			doc.WriteString(formatRow(styleBold.Render(keyVersions), strings.Join(versions, ", "), maxKeyLength, s.width))
+		}
+
+		if len(pkg.Vulnerabilities) > 0 {
+			var (
+				critical = pkg.Vulnerabilities["Critical"]
+				high     = pkg.Vulnerabilities["High"]
+				medium   = pkg.Vulnerabilities["Medium"]
+				low      = pkg.Vulnerabilities["Low"]
+			)
+
+			vunlSummary := []string{}
+			if critical > 0 {
+				vunlSummary = append(vunlSummary, styleRed.Render(locale.Tl("search_critical", "{{.V0}} Critical", strconv.Itoa(critical))))
+			}
+			if high > 0 {
+				vunlSummary = append(vunlSummary, styleOrange.Render(locale.Tl("search_high", "{{.V0}} High", strconv.Itoa(high))))
+			}
+			if medium > 0 {
+				vunlSummary = append(vunlSummary, styleYellow.Render(locale.Tl("search_medium", "{{.V0}} Medium", strconv.Itoa(medium))))
+			}
+			if low > 0 {
+				vunlSummary = append(vunlSummary, styleMagenta.Render(locale.Tl("search_low", "{{.V0}} Low", strconv.Itoa(low))))
+			}
+
+			if len(vunlSummary) > 0 {
+				doc.WriteString(formatRow(styleBold.Render(keyVulns), strings.Join(vunlSummary, ", "), maxKeyLength, s.width))
+			}
+		}
+
+		doc.WriteString("\n")
+	}
+	return doc.String()
+}
+
+func (s structuredSearchResults) MarshalStructured(_ output.Format) interface{} {
+	return s
+}
+
+func formatRow(key, value string, maxKeyLength, width int) string {
+	rowStyle := lipgloss.NewStyle().Width(width)
+
+	// Pad key and wrap value
+	// The viewport does not support padding so we need to pad the key manually
+	paddedKey := strings.Repeat(" ", leftPad) + key + strings.Repeat(" ", maxKeyLength-len(key))
+	valueStyle := lipgloss.NewStyle().Width(width - len(paddedKey))
+
+	wrapped := valueStyle.Render(value)
+
+	// The rendered line ends up being a bit too long, so we need to reduce the
+	// width that we are working with to ensure that the wrapped value fits
+	indentedValue := strings.ReplaceAll(wrapped, "\n", "\n"+strings.Repeat(" ", len(paddedKey)-8))
+
+	formattedRow := fmt.Sprintf("%s%s", paddedKey, indentedValue)
+	return rowStyle.Render(formattedRow) + "\n"
+}
```

Comment:
This is a bit hard to follow, could you document the function, maybe show an example?

Reply:
Tried to add a few more comments. I agree it's not easy to follow. If this batch doesn't help I can try to revisit it again.